test/TensorNetwork_test.jl

```diff
@@ -263,6 +265,20 @@
             replace!(tn, A => new_tensor)
             @test new_tensor === tensors(tn, 1)
             @test B === tensors(tn, 2)
+
+            # Test chain of replacements
+            A = Tensor(zeros(2, 2), (:i, :j))
+            B = Tensor(zeros(2, 2), (:j, :k))
+            C = Tensor(zeros(2, 2), (:k, :l))
+            tn = TensorNetwork([A, B, C])
+
+            @test_throws ArgumentError replace!(tn, A => B, B => C, C => A)
```

Comment:
This should work.

Reply:
Why? We talked that the `replace` function just replaces tensors with the same `labels`.

Reply:
You're right.